src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -61,7 +61,10 @@ public function onKernelRequest(RequestEvent $event)
         if ($request->hasSession()) {
             // no-op
         } elseif (method_exists($request, 'setSessionFactory')) {
-            $request->setSessionFactory(function () { return $this->getSession(); });
+            $session = null;
+            $request->setSessionFactory(function () use (&$session) {
+                return $session ?? $session = $this->getSession();
```

Comment:
this is a bug: The closure is cloned in sub-request leading to multiple call to `$this->getSession()` when oth subRequest and request need the session. This is not a issue in `SessionListener` because it always returns the `session` service. But it became an issue in `FactoryListener` which create a new instance each time.